binding.cc

```diff
@@ -442,19 +446,18 @@ namespace zmq {
 
         // get our next frame it may have the target address and safely copy to our buffer
         zmq_msg_init (&msg2);
-        if (zmq_msg_more(&msg1) == 0) {
+        if (zmq_msg_more(&msg1) == 0 || zmq_recvmsg (s->monitor_socket_, &msg2, 0) == -1) {
           NanThrowError(ExceptionFromError());
-          return;
-        }
-        if (zmq_recvmsg (s->monitor_socket_, &msg2, 0) == -1) {
-          NanThrowError(ExceptionFromError());
-          return;
+          zmq_msg_close(&msg2);
+          continue;
```

Comment:
@ronkorving My intention was to keep on the original behavior, see here: https://github.com/JustinTulloss/zeromq.node/blob/master/binding.cc#L446
It throws and return, now my testing shows that the "throw" doesn't really do anything, and then `UV_MonitorCallback` will be called again reading the next message (because there is a repeating timer there). So my implementation would throw (which actually does nothing), and continue to the next message...

@kkoopa I don't see anything at line 484... I break from the loop if no message received from the socket, this is to keep with the current behavior.
See: https://github.com/JustinTulloss/zeromq.node/blob/master/binding.cc#L475
If no message received there is a call to `uv_timer_stop`, stopping the monitor interval.
So I do the same here: https://github.com/ValYouW/zeromq.node/blob/master/binding.cc#L479
Set `reload_timer=false` and break from the while loop, this will cause to the monitor timer not to be reset, see: https://github.com/ValYouW/zeromq.node/blob/master/binding.cc#L485

Regarding the `NaNThrowError`, I think that for this to really throw it should be `return NaNThrowError`, now do we really want to throw? isn't that a recoverable situation? (like ignore and move to the next message??) I don't really now...